titus-common/src/main/java/com/netflix/titus/common/util/CollectionsExt.java

```diff
@@ -289,6 +289,14 @@ private CollectionsExt() {
         return result;
     }
 
+    public static <K, U, V, M extends Map<K, U>>
+    Map<K, V> mapValues(M source, Function<U, V> valueMapper, Supplier<Map<K, V>> mapSupplier) {
```

Comment:
Why `M` not `Map<K, U>`?

Reply:
you are right, fixing. I wanted it being able to return the concrete impl when needed (I was using `LinkedHashMap` in a previous version of this, but it is not needed anymore.